modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/template/test/TestTemplateContextContributor.java

```diff
@@ -0,0 +1,42 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.notification.internal.template.test;
+
+import com.liferay.portal.kernel.template.TemplateContextContributor;
+
+import java.io.Serializable;
+
+import java.util.Map;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * @author Richard Jeremias
+ */
+@Component(
+	property = "type=" + TemplateContextContributor.TYPE_GLOBAL,
+	service = TemplateContextContributor.class
+)
+public class TestTemplateContextContributor
+	implements TemplateContextContributor {
+
+	@Override
+	public void prepare(
+		Map<String, Object> contextObjects,
+		HttpServletRequest httpServletRequest) {
+
+		contextObjects.put(
+			"templateComponent", new InnerTemplateContextContributor());
```

Comment:
@rijema why use the `@Component` annotation on the inner class when the constructor could be used directly?

Reply:
Even though we're explicitly using the constructor, to simulate the customer scenario I still need to inform OSGI this class is a service, registering on its list. To conclude the test, I could injected/reference the service, or diminish the changes I could just instantiate. Whichever option was chosen, I would have access to the service from the template.

Reply:
@rijema to me it doesn't make sense to use both a constructor and the `@Component` annotation at the same time. 

Did you run the test without your fix to make sure it fails?

Reply:
Yes. I ran the the test before submitting the changes. The both annotations were added to give the customer context. Indeed, if we want to resume the test to send the reference of only one service we can do something like 

```
@Component(
	property = "type=" + TemplateContextContributor.TYPE_GLOBAL,
	service = TemplateContextContributor.class
)
public class PlaceCubeImpl
	implements TemplateContextContributor {

	@Override
	public void prepare(
		Map<String, Object> contextObjects,
		HttpServletRequest httpServletRequest) {

		contextObjects.put("objectObject", new Serializable(){});
	}

}

```
But the test would have as response something like com.liferay.notification.internal.handler.TestTemplateContextContributor$1@750429ef and not com.liferay.notification.internal.handler.TestTemplateContextContributor$InnerTemplateContextContributor@4ae34f16. Is it ok to validate the minimal?  Thanks!!